### ISSUE TITLE
Google Analytics event displays wrong active page in real-time stats

### ISSUE THREAD

**Issue by philhudson91** [luisfarzati/angulartics#340]
Google Analytics event displays wrong active page in real-time stats

To replicate - fire the event on a sub-view in Angular (e.g. not /#/, but /#/page). Event triggers real-time active page as /.

**Comment by jcsmesquita** [comment]
+1

**Comment by philhudson91** [comment]
Any ideas with this please? It's really affecting my analytics!

**Comment by johanhar** [comment]
seems to be the same problem with revolunet/angular-google-analytics.

**Comment by philhudson91** [comment]
I've fixed it, check my blog post here:

http://www.phil-hudson.com/how-to-fix-angularitics-event-wrong-active-page-in-real-time-stats/

Basically I needed to send the path with the page param of the event object.

**Comment by timelf123** [comment]
For posterity, the fix is including `page: $location.path()` in the event tracking call. I'm opening this one back up for testing

`$analytics.eventTrack(time, {category: 'Categoryname', label: 'LabelName', page: $location.path()} }`

**Comment by philhudson91** [comment]
Cheers, I should've pasted the code in my post!

**Comment by DanWilkerson** [comment]
Addressed in #410 for GTM.

**Comment by ProLoser** [comment]
Are you using ui-router? i believe the path is being calculated incorrectly for the ui-router edge-case.

**Comment by ProLoser** [comment]
Specifically these 2 lines: https://github.com/angulartics/angulartics/blob/master/src/angulartics.js#L195-L196 as `$location` is not updated at the time of `$stateChangeSuccess` so the current page URL needs to be calculated from `$state` service. This will cause a 1-page lag for all reported pageTrack events. I have a local fix but have yet to push a fix to the lib.

**Comment by DanWilkerson** [comment]
@ProLoser No, but this is for Google Tag Manager, which sits on top of all this.

**Comment by timelf123** [comment]
@ProLoser can you roll a PR for the ui-router fix (unless I missed that already)?

**Comment by ProLoser** [comment]
https://github.com/angulartics/angulartics/commit/543da16133a878715afc4432435dc9fd3a196a79

I think it's already done
